src/TPPCore.Service.ChatLogger/ChatLoggerService.cs

```diff
@@ -1,30 +1,34 @@
-﻿using TPPCore.Service.Common;
+﻿using TPPCore.Service.Chat;
+using TPPCore.Service.Common;
 
 namespace TPPCore.Service.ChatLogger
 {
-	public class ChatLoggerService : IService
-	{
-		private ServiceContext context;
+    public class ChatLoggerService : IService
+    {
+        private ServiceContext context;
+        private bool running;
 
-		public void Initialize(ServiceContext context)
-		{
-			this.context = context;
+        public void Initialize(ServiceContext context)
+        {
+            this.context = context;
+            running = true;
+            LogManager.Configure(this.context);
 
-			LogManager.Configure(this.context);
+            context.PubSubClient.Subscribe(ChatTopics.Raw,
+                (topic, message) =>
+                {
+                    LogManager.LogMessage(message);
+                });
+        }
 
-			context.PubSubClient.Subscribe(ChatLoggerTopics.Raw,
-				(topic, message) =>
-				{
-					LogManager.LogMessage(message);
-				});
-		}
+        public void Run()
+        {
+            while (running) ;
```

Comment:
This works but you should use a synchronization primitive to put the thread to sleep instead of busy waiting. You could use `ManualResetEvent` to wait here and then in the `shutdown` method, call `set` to wake it back up.